### DISCUSSION TITLE
Receptor occupancy for competitive binding

### DISCUSSION THREAD

**Discussion by wilbertdew** [Open-Systems-Pharmacology/Forum#1465] [Feature Requests & Ideas]
Hi All,

With the current implementation in PK-Sim (v11.0), the calculated receptor occupancy in the default receptor occupancy observer does not take multiple compounds binding to the same receptor into account. The applied formula is "Protein + Complex > 0 ? Complex/(Protein + Complex) : 0", where Protein is the free receptor and Complex the complex with the molecule for which the occupancy is calculated. With competitive binding, this should be Complex/Total, where Total = Protein + Complex + complexes to competitive binders. I guess this is difficult to implement in such a generic way that it detects the total target even in MoBi where free naming and reaction modification occurs, but maybe it's more straightforward in PK-Sim? Otherwise a flag in case of competitive binding or renaming of the observer might be helpful.

**Comment by Tan-Zas** [comment]
Hi all,

I’d like to revive this discussion and provide a more detailed explanation.

The calculation of the occupancy is not accurate in the case of competitive binding. The default observer accounts only for one drug-target complex, while we have, e.g., two complexes affecting the target occupancy.  In other words, the default observer calculates occupancy using the formula:

 
<img width="599" height="112" alt="image" src="https://github.com/user-attachments/assets/af95a59b-919a-45bf-9bce-525913db51a9" />

 while the occupancy should be calculated as the bound receptor divided by the total receptor amount:

 
<img width="940" height="115" alt="image" src="https://github.com/user-attachments/assets/594040e0-d30c-4df6-8939-de1183d75b63" />

where the denominator is the total receptor, including all the receptor states.

 

The default calculation of the target occupancy in the case with several competing ligands misses the contribution of receptors bound by other ligands (e.g., bound_by_ligandB). As a result, it only considers ligandA and the free receptor, leading to an incomplete representation of total receptor occupancy.

**Comment by PavelBal** [comment]
@wilbertdew @Tan-Zas If you work out a concept how this could be implemented, we can do it.

Spontaneous thoughts - use tags and sum formulas. Each complex of protein `A` could get a tag `A-Complex`, each ligand could get a tag `A-Ligand`. The same could apply to products of reactions, e.g. to calculate total fraction metabolized.

Then use sum formula to calculate `Total amount A-Lingads` etc.

**Comment by wilbertdew** [comment]
@Pavel Balazki ***@***.***> yes something like that would work
I guess, Vanessa also suggested that. Alternatively, for the protein
binding processes, the complexes have automatically generated names, so
that would allow to also adapt the formula automatically in PK-Sim. If we
want it to be also appropriate for any new binding reaction that is added
in MoBi, its probably nicer to have the tag approach I guess? We would only
need to keep track of the complexes btw for the occupancy formula, not the
ligands.

Op do 11 sep 2025 om 11:26 schreef Pavel Balazki ***@***.***>:
